profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java

```diff
@@ -1435,6 +1438,440 @@ public void testSubscriptionEventsWithCatalogChange() throws Exception {
+        clock.setDay(initialDate);
+
+        final Account accountJson = createAccountWithDefaultPaymentMethod();
+
+        final Subscription input = new Subscription();
+        input.setAccountId(accountJson.getAccountId());
+        input.setProductName("Shotgun");
+        input.setProductCategory(ProductCategory.BASE);
+        input.setBillingPeriod(BillingPeriod.MONTHLY);
+        input.setPriceList(PriceListSet.DEFAULT_PRICELIST_NAME);
+        final Subscription entitlementJson = subscriptionApi.createSubscription(input,
+                                                                                initialDate,
+                                                                                null,
+                                                                                false,
+                                                                                false,
+                                                                                false,
+                                                                                true,
+                                                                                DEFAULT_WAIT_COMPLETION_TIMEOUT_SEC,
+                                                                                NULL_PLUGIN_PROPERTIES,
+                                                                                requestOptions);
+
+        Assert.assertEquals(entitlementJson.getState(), EntitlementState.ACTIVE);
+        Assert.assertEquals(internalCallContext.toLocalDate(entitlementJson.getBillingStartDate()), initialDate);
+        Assert.assertEquals(internalCallContext.toLocalDate(entitlementJson.getStartDate()), initialDate);
+        Assert.assertEquals(entitlementJson.getProductName(), input.getProductName());
+
+        final LocalDate changeDate = clock.getUTCToday().plusDays(5);
+
+        final Subscription newSubscriptionPlan = new Subscription();
+        newSubscriptionPlan.setSubscriptionId(entitlementJson.getSubscriptionId());
+        newSubscriptionPlan.setPlanName("pistol-monthly");
+
+        subscriptionApi.changeSubscriptionPlan(entitlementJson.getSubscriptionId(), newSubscriptionPlan, changeDate, true, DEFAULT_WAIT_COMPLETION_TIMEOUT_SEC, null, NULL_PLUGIN_PROPERTIES, requestOptions);
+
+        // Retrieves with GET
+        Subscription subscription = subscriptionApi.getSubscription(entitlementJson.getSubscriptionId(), requestOptions);
+        Assert.assertEquals(subscription.getState(), EntitlementState.ACTIVE);
+        Assert.assertEquals(entitlementJson.getProductName(), input.getProductName()); //still old plan since changeDate is not reached
+
+        clock.setDay(changeDate);
+        subscription = subscriptionApi.getSubscription(entitlementJson.getSubscriptionId(), requestOptions);
+        Assert.assertEquals(newSubscriptionPlan.getPlanName(), subscription.getPlanName()); //now on new plan
+
+        //TODO_1375 - Is there some way of verifying the change dates?
```

Comment:
You might be able to get that information via the billing timeline API (https://killbill.github.io/slate/index.html#account-retrieve-account-timeline).

Reply:
Thanks! Instead of retrieving the entire account timeline, I've now fixed it by retrieving the subscription events [here](https://github.com/reshmabidikar/killbill/blob/5ded768eaf13082d107eec92d3fdc7c7ed716392/profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java#L1701). LMK if you think this will not work.